raster/src/main/scala/geotrellis/raster/io/geotiff/ByteConstantNoDataGeoTiffSegment.scala

```diff
@@ -0,0 +1,102 @@
+package geotrellis.raster.io.geotiff
+
+import geotrellis.raster._
+import geotrellis.raster.io.geotiff.utils._
+import geotrellis.raster.io.geotiff.compression._
+
+import java.util.BitSet
+
+import spire.syntax.cfor._
+
+class ByteConstantNoDataCellTypeGeoTiffSegment(bytes: Array[Byte]) extends ByteGeoTiffSegment(bytes) {
+  def getInt(i: Int): Int = b2i(get(i))
+  def getDouble(i: Int): Double = b2i(get(i))
+
+  protected def intToByteOut(v: Int): Byte = i2b(v)
+  protected def doubleToByteOut(v: Double): Byte = d2b(v)
+
+  protected def convertToConstantNoData(cellType: DataType with ConstantNoData): Array[Byte] =
+    cellType match {
+      case ByteConstantNoDataCellType =>
+        val arr = Array.ofDim[Byte](size)
+        cfor(0)(_ < size, _ + 1) { i => arr(i) = get(i) }
```

Comment:
Is this just converting to the same type? If so we should just pass it through and not copy.